src/main/java/core/basesyntax/RightTriangle.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class RightTriangle extends Figure {
+    private static final int NUMBER = 10;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
```

Comment:
The same goes for ColorSupplier - it's not rightTriangle job to set itself a color. It just should be able to take the color from the constructor or setter and change it accordingly. The reason and nature of generation of these parameters are out of the scope of the class itself